lib/stats/device.go

```diff
@@ -10,26 +10,34 @@ import (
 	"time"
 
 	"github.com/syncthing/syncthing/lib/db"
+	"github.com/syncthing/syncthing/lib/db/backend"
+	"github.com/syncthing/syncthing/lib/protocol"
+)
+
+const (
+	lastSeenKey     = "lastSeen"
+	connDurationKey = "lastConnDuration"
 )
 
 type DeviceStatistics struct {
-	LastSeen time.Time `json:"lastSeen"`
+	LastSeen                time.Time `json:"lastSeen"`
+	LastConnectionDurationS float64   `json:"lastConnectionDurationS"`
```

Comment:
A `time.Duration` marshals as an integer nanosecond count which isn't super intuitive on the receiving API side, so going with seconds here instead.

Reply:
Why not handle it as time.Duration properly?
Feels confusing seeing .Seconds() and .Nanoseconds() in the same thing.

Reply:
We do need subsecond precision, because I want to tell when a connection is closed immediately due to other side having paused us, for example. That's zero seconds, which would otherwise be impossible to differentiate from the "no value" zero. (We could use `-1*time.Second` or some sentinel for that but that's also ugly.)

I made it seconds here instead of time.Duration because time.Duration marshals to for example 169461293376 while I'd much rather see 169.461293376 in the API output...